lib_eio/path.ml

```diff
@@ -30,10 +30,21 @@ let with_lines path fn =
   let buf = Buf_read.of_flow flow ~max_size:max_int in
   fn (Buf_read.lines buf)
 
-let load path =
-  with_open_in path @@ fun flow ->
-  let buf = Buf_read.of_flow flow ~max_size:max_int in
-  Buf_read.take_all buf
+let load ((t:#Fs.dir), path) =
+  with_open_in (t, path) @@ fun flow ->
```

Comment:
```suggestion
let load t =
  with_open_in t @@ fun flow ->
```

Reply:
This isn't resolved. Keeping it as one will also fix the problem that the string in the error loses the dir tag, since you can then use `Path.pp`.

(we might need a bit of variable renaming in this module; lots of functions here are using `t` for the directory and `path` for a full path, which is now a bit backwards)

Reply:
I'm a bit confused, we don't really include the dir in the error string for any of the Eio.Fs exceptions:
```
let open_in ~sw ((t:#Fs.dir), path) = t#open_in ~sw path

  method open_in ~sw path =
    let fd = File.open_ ~sw (self#resolve path) [`NOFOLLOW; `RDONLY] |> or_raise_path path in
    (flow fd :> <Eio.File.ro; Eio.Flow.close>)                                        ^^^^
```
```
utop # let _ = Eio_luv.run (fun env -> let cwd = Eio.Stdenv.cwd env in Eio.Switch.run @@ fun sw -> Eio.Path.load Eio.Path.(cwd / "not_there"));;
Exception: Eio__Fs.Not_found ("not_there", Eio_luv.Luv_error 877502405).

let _ = Eio_linux.run (fun env -> let cwd = Eio.Stdenv.cwd env in Eio.Switch.run @@ fun sw -> Eio.Path.load Eio.Path.(cwd / "not_there"));;
Exception:
Eio__Fs.Not_found ("not_there", Unix.Unix_error (Unix.ENOENT, "openat2", "")).
```

Reply:
OK, let's sort that out later then! It just seemed a shame to be splitting the path up only for the purpose of producing a slightly less informative error message.